packages/extractor-vue/src/vue-extractor.ts

```diff
@@ -0,0 +1,28 @@
+import * as vueCompiler from "@vue/compiler-sfc"
+// @ts-expect-error issue with typings, will be fixed shortly
+import babel from "@lingui/cli/api/extractors/babel"
+import type { ExtractorCtx, ExtractorType } from "@lingui/conf"
+
+export const vueExtractor: ExtractorType = {
+  match(filename: string) {
+    return filename.endsWith(".vue")
+  },
+  extract(
+    filename: string,
+    code: string,
+    onMessageExtracted,
+    ctx: ExtractorCtx
+  ) {
+    const {
+      descriptor: { script, scriptSetup },
+    } = vueCompiler.parse(code, { sourceMap: true, filename })
```

Comment:
What was the issue to add a template for extraction as well?

Reply:
> What was the issue to add a template for extraction as well?

Could probably make that work as well. It's just a bit more intricate than the script portion of vue files.

Reply:
Also not sure if that is good practice/works in Vue. I'm not really a Vue developer myself, so my knowledge on that area is a bit lacking.

Reply:
https://github.com/formatjs/formatjs/blob/main/packages/cli-lib/src/vue_extractor.ts#L72-L95